### DISCUSSION TITLE
Mantine focus styling not consistent and customizable?

### DISCUSSION THREAD

**Discussion by maxratajczak** [mantinedev/mantine#2789] [Feedback]
First off, I'd like to say I love Mantine, I'm using it for my project and it's amazing. Hats off to @rtivital and all the other contributors, you've done amazing work!

I am having some trouble with customizing the focus styling though. Maybe I'm approaching it wrong? But I can't find much in the docs about focus styling. For example, the outline thickness is bigger on the Chips component than other components such as an input.

<img width="462" alt="Screen Shot 2022-10-24 at 5 34 39 PM" src="https://user-images.githubusercontent.com/75450072/197634068-df5b0607-70a3-4555-a87c-360a7f89a465.png">
<img width="555" alt="Screen Shot 2022-10-24 at 5 34 52 PM" src="https://user-images.githubusercontent.com/75450072/197634129-69667ea3-b222-447d-a12c-ba817231fb9d.png">

Debugging this is also kind of weird because when I use chrome tools to activate different states on the chips, none of the focus states activate the thick light blue border that the keyboard activates.

I'm looking to have a global style for all component focus & focus-within, but setting this in a global css sheet doesn't work as mantine style override this. I've also tried messing around in the globalStyles property in the MantineProvider and again, some components aren't affected such as the Chip components and the default mantine styles override these styles.

<img width="874" alt="Screen Shot 2022-10-24 at 5 47 50 PM" src="https://user-images.githubusercontent.com/75450072/197636100-e8138c14-ad5f-45e1-93c8-7d7707aa6283.png">

Focus styling should be consistent through an application and having 3 different styles (mantine thin blue, mantine thick blue & chrome default dark blue) definitely has an effect on the UI/UX.

Personally, I'd love to see an option to disable mantine focus styling so we're able to implement our own. Maybe something similar to the focusRing props in the theme object where it will just disable mantine focus styles.

**Comment by rtivital** [comment]
Mantine uses 2 focus ring styles – one used for inputs, another one for everything else. None of the components have default browser focus styles, if you encounter it, please open an issue, and it will be fixed. You can override focus styles on theme for inputs and other elements – https://codesandbox.io/s/amazing-fast-n7edl5?file=/src/App.tsx

**Reply by maxratajczak** [reply]
Oh awesome, thank you this helps a lot. But I still can't seem to change any styling on the Chip component, that seems to be the only one that isn't affected, any idea why?

**Reply by rtivital** [reply]
You can view component source code to find out which styles should be applied – https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Chip/Chip.styles.ts#L137:L165

**Reply by maxratajczak** [reply]
The styles still cannot be overridden even when changing the boxShadow on focus. You aren't even able to debug in chrome dev tools. Something's up.
